client/state/shortcodes/actions.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+import {
+	SHORTCODE_FETCH,
+	SHORTCODE_RECEIVE
+} from 'state/action-types';
+
+export function fetchShortcode( siteId, shortcode ) {
+	return ( dispatch ) => {
+		dispatch( {
+			type: SHORTCODE_FETCH,
+			siteId,
+			shortcode
+		} );
+
+		return wpcom.undocumented().site( siteId ).shortcodes( {
+			shortcode: shortcode
```

Comment:
ES2015 includes a nice shorthand for creating objects where assigning a key to the same variable name:

```js
{ shortcode }
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#New_notations_in_ECMAScript_2015